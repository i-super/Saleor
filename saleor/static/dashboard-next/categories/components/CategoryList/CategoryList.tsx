import Button from "@material-ui/core/Button";
import Card from "@material-ui/core/Card";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import ListSubheader from "@material-ui/core/ListSubheader";
import Folder from "@material-ui/icons/Folder";
import * as React from "react";

import CardTitle from "../../../components/CardTitle";
import Skeleton from "../../../components/Skeleton";
import i18n from "../../../i18n";

interface CategoryListProps {
  categories?: Array<{
    id: string;
    name: string;
  }>;
  displayTitle?: boolean;
  onAdd?();
  onRowClick?(id: string): () => void;
}

const CategoryList: React.StatelessComponent<CategoryListProps> = ({
  categories,
  displayTitle,
  onAdd,
  onRowClick
}) => (
  <Card>
    {displayTitle && (
      <CardTitle
        title={i18n.t("Categories")}
        toolbar={
          <Button color="secondary" variant="flat" onClick={onAdd}>
            {i18n.t("Add category")}
          </Button>
        }
      />
    )}
    <List>
      {categories === undefined || categories === null ? (
        <ListItem>
          <ListItemIcon>
            <Folder />
          </ListItemIcon>
          <ListItemText>
            <Skeleton />
          </ListItemText>
        </ListItem>
      ) : categories.length > 0 ? (
        categories.map(category => (
          <ListItem
            button={!!onRowClick}
            key={category.id}
            onClick={!!onRowClick ? onRowClick(category.id) : undefined}
          >
            <ListItemIcon>
              <Folder />
            </ListItemIcon>
            <ListItemText primary={category.name} />
          </ListItem>
        ))
      ) : (
        <ListSubheader>{i18n.t("No categories found")}</ListSubheader>
      )}
    </List>
  </Card>
);
export default CategoryList;
