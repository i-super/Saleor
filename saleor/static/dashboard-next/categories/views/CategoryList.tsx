import * as React from "react";

import { categoryAddUrl, categoryUrl } from "..";
import Navigator from "../../components/Navigator";
import { maybe } from "../../misc";
import { CategoryListPage } from "../components/CategoryListPage/CategoryListPage";
import { TypedRootCategoriesQuery } from "../queries";

export const CategoryList: React.StatelessComponent = () => (
  <Navigator>
    {navigate => (
      <TypedRootCategoriesQuery>
        {({ data }) => (
          <CategoryListPage
            categories={maybe(
              () => data.categories.edges.map(edge => edge.node),
              []
            )}
            onAddCategory={() => navigate(categoryAddUrl())}
            onCategoryClick={id => () =>
              navigate(categoryUrl(encodeURIComponent(id)))}
          />
        )}
      </TypedRootCategoriesQuery>
    )}
  </Navigator>
);
export default CategoryList;
