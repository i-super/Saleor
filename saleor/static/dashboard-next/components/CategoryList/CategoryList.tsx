import Folder from "@material-ui/icons/Folder";
import List, {
  ListItem,
  ListItemIcon,
  ListItemText,
  ListSubheader
} from "material-ui/List";
import * as React from "react";

import { CategoryPropertiesQuery } from "../../gql-types";
import i18n from "../../i18n";
import Skeleton from "../Skeleton";

interface CategoryListProps {
  categories?: Array<{
    id: string;
    name: string;
  }>;
  onClick?(id: string);
}
export const CategoryList: React.StatelessComponent<CategoryListProps> = ({
  categories,
  onClick
}) => (
  <List>
    {categories === undefined || categories === null ? (
      <ListItem>
        <ListItemIcon>
          <Folder />
        </ListItemIcon>
        <ListItemText>
          <Skeleton />
        </ListItemText>
      </ListItem>
    ) : categories.length > 0 ? (
      categories.map(category => (
        <ListItem
          button={!!onClick}
          key={category.id}
          onClick={
            onClick !== undefined ? () => onClick(category.id) : undefined
          }
        >
          <ListItemIcon>
            <Folder />
          </ListItemIcon>
          <ListItemText primary={category.name} />
        </ListItem>
      ))
    ) : (
      <ListSubheader>{i18n.t("No categories found")}</ListSubheader>
    )}
  </List>
);

export default CategoryList;
