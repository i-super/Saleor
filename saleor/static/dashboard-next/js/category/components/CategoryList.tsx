import Folder from "material-ui-icons/Folder";
import Button from "material-ui/Button";
import Grid from "material-ui/Grid";
import List, {
  ListItem,
  ListItemIcon,
  ListItemText,
  ListSubheader
} from "material-ui/List";
import Typography from "material-ui/Typography";
import * as React from "react";
import { Link } from "react-router-dom";

import Skeleton from "../../components/Skeleton";
import { CategoryPropertiesQuery } from "../../gql-types";
import i18n from "../../i18n";

interface CategoryListProps {
  categories?: CategoryPropertiesQuery["category"]["children"]["edges"];
  onClick?(id: string);
}
export const CategoryList: React.StatelessComponent<CategoryListProps> = ({
  categories,
  onClick
}) => (
  <List>
    {categories === undefined ? (
      <ListItem>
        <ListItemIcon>
          <Folder />
        </ListItemIcon>
        <ListItemText>
          <Skeleton />
        </ListItemText>
      </ListItem>
    ) : categories.length > 0 ? (
      categories.map(edge => (
        <ListItem
          button={!!onClick}
          key={edge.node.id}
          onClick={
            onClick !== undefined ? () => onClick(edge.node.id) : undefined
          }
        >
          <ListItemIcon>
            <Folder />
          </ListItemIcon>
          <ListItemText primary={edge.node.name} />
        </ListItem>
      ))
    ) : (
      <ListSubheader>{i18n.t("No categories found")}</ListSubheader>
    )}
  </List>
);

export default CategoryList;
