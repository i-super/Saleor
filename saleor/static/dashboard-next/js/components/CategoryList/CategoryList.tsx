import Folder from "material-ui-icons/Folder";
import List, {
  ListItem,
  ListItemIcon,
  ListItemText,
  ListSubheader
} from "material-ui/List";
import * as React from "react";

import { CategoryPropertiesQuery } from "../../gql-types";
import i18n from "../../i18n";
import Skeleton from "../Skeleton";

interface CategoryListProps {
  categories?: CategoryPropertiesQuery["category"]["children"]["edges"];
  onClick?(id: string);
}
export const CategoryList: React.StatelessComponent<CategoryListProps> = ({
  categories,
  onClick
}) => (
  <List>
    {categories === undefined ? (
      <ListItem>
        <ListItemIcon>
          <Folder />
        </ListItemIcon>
        <ListItemText>
          <Skeleton />
        </ListItemText>
      </ListItem>
    ) : categories.length > 0 ? (
      categories.map(edge => (
        <ListItem
          button={!!onClick}
          key={edge.node.id}
          onClick={
            onClick !== undefined ? () => onClick(edge.node.id) : undefined
          }
        >
          <ListItemIcon>
            <Folder />
          </ListItemIcon>
          <ListItemText primary={edge.node.name} />
        </ListItem>
      ))
    ) : (
      <ListSubheader>{i18n.t("No categories found")}</ListSubheader>
    )}
  </List>
);

export default CategoryList;
