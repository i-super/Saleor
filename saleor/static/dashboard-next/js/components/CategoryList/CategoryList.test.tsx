import * as React from "react";
import { MemoryRouter } from "react-router-dom";
import * as renderer from "react-test-renderer";

import CategoryList from "./CategoryList";

const categories = [
  {
    node: {
      id: "cat1",
      name: "Apparel"
    }
  },
  {
    node: {
      id: "cat2",
      name: "Accessories"
    }
  },
  {
    node: {
      id: "cat3",
      name: "Groceries"
    }
  },
  {
    node: {
      id: "cat4",
      name: "Books"
    }
  }
];

describe("<CategoryList />", () => {
  it("renders while data is loading", () => {
    const component = renderer.create(
      <MemoryRouter>
        <CategoryList />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
  it("renders when data is fully loaded", () => {
    const component = renderer.create(
      <MemoryRouter>
        <CategoryList categories={categories} onClick={jest.fn()} />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
  it("renders when category list is empty", () => {
    const component = renderer.create(
      <MemoryRouter>
        <CategoryList categories={[]} />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
