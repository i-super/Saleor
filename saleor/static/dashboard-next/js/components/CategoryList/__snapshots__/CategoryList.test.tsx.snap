// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryList /> renders when category list is empty 1`] = `
<ul
  className="MuiList-root-1 MuiList-padding-2"
>
  <li
    className="MuiListSubheader-root-63 MuiListSubheader-sticky-67"
  >
    No categories found
  </li>
</ul>
`;

exports[`<CategoryList /> renders when data is fully loaded 1`] = `
<ul
  className="MuiList-root-1 MuiList-padding-2"
>
  <div
    className="MuiButtonBase-root-53 MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-12 MuiListItem-button-13"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-16 MuiListItemIcon-root-15"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
      />
    </svg>
    <div
      className="MuiListItemText-root-22"
    >
      <h3
        className="MuiTypography-root-28 MuiTypography-subheading-35 MuiListItemText-primary-25"
      >
        Apparel
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-56"
    />
  </div>
  <div
    className="MuiButtonBase-root-53 MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-12 MuiListItem-button-13"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-16 MuiListItemIcon-root-15"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
      />
    </svg>
    <div
      className="MuiListItemText-root-22"
    >
      <h3
        className="MuiTypography-root-28 MuiTypography-subheading-35 MuiListItemText-primary-25"
      >
        Accessories
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-56"
    />
  </div>
  <div
    className="MuiButtonBase-root-53 MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-12 MuiListItem-button-13"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-16 MuiListItemIcon-root-15"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
      />
    </svg>
    <div
      className="MuiListItemText-root-22"
    >
      <h3
        className="MuiTypography-root-28 MuiTypography-subheading-35 MuiListItemText-primary-25"
      >
        Groceries
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-56"
    />
  </div>
  <div
    className="MuiButtonBase-root-53 MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-12 MuiListItem-button-13"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-16 MuiListItemIcon-root-15"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
      />
    </svg>
    <div
      className="MuiListItemText-root-22"
    >
      <h3
        className="MuiTypography-root-28 MuiTypography-subheading-35 MuiListItemText-primary-25"
      >
        Books
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-56"
    />
  </div>
</ul>
`;

exports[`<CategoryList /> renders while data is loading 1`] = `
<ul
  className="MuiList-root-1 MuiList-padding-2"
>
  <li
    className="MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-12"
    disabled={false}
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-16 MuiListItemIcon-root-15"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
      />
    </svg>
    <div
      className="MuiListItemText-root-22"
    >
      <h3
        className="MuiTypography-root-28 MuiTypography-subheading-35 MuiListItemText-primary-25"
      >
        <span
          className="Component-skeleton-52"
          style={undefined}
        >
          â€Œ
        </span>
      </h3>
    </div>
  </li>
</ul>
`;
